wallet/walletrpc.c

```diff
@@ -738,13 +737,21 @@ static struct command_result *json_signpsbt(struct command *cmd,
 	struct wally_psbt *psbt, *signed_psbt;
 	struct utxo **utxos;
 	u32 *input_nums;
+	u32 psbt_version;
 
 	if (!param(cmd, buffer, params,
 		   p_req("psbt", param_psbt, &psbt),
 		   p_opt("signonly", param_input_numbers, &input_nums),
 		   NULL))
 		return command_param_failed();
 
+	/* We internally deal with v2 only but we want to return V2 if given */
+	psbt_version = psbt->version;
+	if (!psbt_set_version(psbt, 2)) {
+		return command_fail(cmd, LIGHTNINGD,
+				    "Could not set PSBT version");
```

Comment:
This should probably print the psbt at least?

Reply:
done